Add 2.15 docs for authenticated private registries (#1235)
rancher/rancher-product-docs: versions/v2.15/modules/en/pages/installation-and-upgrade/other-installation-methods/air-gapped/install-rancher-ha.adoc

| `<REGISTRY.YOURDOMAIN.COM:PORT>`
| Configure Rancher server to always pull from your private registry when provisioning clusters.

| `systemRegistryInheritsPullSecrets`
| `<boolean>`

> | Sets the default value of the `system-default-registry-pull-secrets` global setting to the same image pull secrets used for the Rancher installation. This can be used when node-level registry configuration is not possible, and when these credentials should be propagated to all downstream clusters by default.

Reviewer: ```suggestion
Sets the default value of the `system-default-registry-pull-secrets` global setting to the image pull secrets used during Rancher installation. Use this option when node-level registry configuration is not possible and you want to propagate these credentials to all downstream clusters by default.
```

The installation is complete.

[#_optional_install_rancher_with_an_authenticated_default_registry]
=== Optional: Install Rancher with an Authenticated Default Registry

> Starting with `v2.15.0`, Rancher is able to leverage an authenticated `systemDefaultRegistry` using Kubernetes image pull secrets. This configuration differs from the top level `imagePullSecrets` and associated `image.repository` values offered, which only impact the primary Rancher deployment. For more detailed information on how Rancher works with authenticated global registries and pull secrets at both the global and cluster levels, refer to {xref-authenticated-private-registries}[this page]

Reviewer: ```suggestion
Starting in Rancher v2.15.0, Rancher supports an authenticated `systemDefaultRegistry` using Kubernetes image pull secrets. This configuration differs from the top-level `imagePullSecrets` and `image.repository` values, which apply only to the primary Rancher deployment. For details on working with global registries and pull secrets across local and downstream clusters, see {xref-authenticated-private-registries}[Authenticated Private Registries].
```

[#_using_an_authenticated_global_registry_with_pull_secrets]
==== Using an Authenticated Global Registry with Pull Secrets

> If your private registry requires authentication, and you cannot configure credentials at the node level, or want to automatically distribute credentials to downstream clusters by default, you can configure both the `system-default-registry` and `system-default-registry-pull-secrets` settings. This configuration will result in the local cluster, and all downstream clusters by default, being configured to pull from the authenticated registry.

Reviewer: ```suggestion
If your private registry requires authentication, you can configure the `system-default-registry` and `system-default-registry-pull-secrets` settings. Use this option when you cannot configure node-level credentials or want to distribute credentials to downstream clusters automatically. This option configures the local cluster and all downstream clusters to pull from the authenticated registry by default.
```

> There are two main approaches to configure default pull secrets this when installing Rancher using Helm:

Reviewer: ```suggestion
You can configure default pull secrets in two ways when installing Rancher with Helm:
```

**Configuration 1: Using the `systemRegistryInheritsPullSecrets` Helm option**

> When set to `true`, Rancher sets the default value of the `system-default-registry-pull-secrets` global setting to the same image pull secrets used for the Rancher installation itself. The setting remains editable post-install via **Global Settings** in the Rancher UI.

Reviewer: ```suggestion
When set to `true`, Rancher sets the default value of the `system-default-registry-pull-secrets` global setting to the image pull secrets used during the Rancher installation. You can edit this setting after installation in **Global Settings** in the Rancher UI.
```

[,plain]
----
helm install rancher ./rancher-<VERSION>.tgz \
  --namespace cattle-system \
  --set hostname=<RANCHER.YOURDOMAIN.COM> \
  --set systemDefaultRegistry=<REGISTRY.YOURDOMAIN.COM:PORT> \
  --set systemRegistryInheritsPullSecrets=true \
  --set useBundledSystemChart=true
----

**Configuration 2: Using the `CATTLE_BASE_REGISTRY_PULL_SECRETS` environment variable**

> This environment variable can be used to set the default value of the `system-default-registry-pull-secrets` global setting to arbitrary pull secret(s) located in the `cattle-system` namespace. The setting remains editable post-install via **Global Settings** in the Rancher UI.

Reviewer: ```suggestion
Use this environment variable to set the default value of the `system-default-registry-pull-secrets` global setting to specific pull secrets in the `cattle-system` namespace. You can edit this setting after installation in **Global Settings** in the Rancher UI.
```

[,plain]
----
helm install rancher ./rancher-<VERSION>.tgz \
  --namespace cattle-system \
  --set hostname=<RANCHER.YOURDOMAIN.COM> \
  --set systemDefaultRegistry=<REGISTRY.YOURDOMAIN.COM:PORT> \
  --set extraEnv[0].name=CATTLE_BASE_REGISTRY_PULL_SECRETS \
  --set extraEnv[0].value='<SECRET1>,<SECRET2>' \
  --set useBundledSystemChart=true
----

[NOTE]
====

> The above configurations set the _default_ values for the `system-default-registry` and `system-default-registry-pull-secrets` settings. Before these settings can be reset to an empty value, the associated helm value (e.g. `systemDefaultRegistry`) or environment variable (e.g. `CATTLE_BASE_REGISTRY_PULL_SECRETS`) must be removed from the Helm installation.

Reviewer: ```suggestion
These configurations set the default values for the `system-default-registry` and `system-default-registry-pull-secrets` settings. To clear these settings, you must remove the associated Helm value (such as `systemDefaultRegistry`) or environment variable (such as `CATTLE_BASE_REGISTRY_PULL_SECRETS`) from your Helm installation.
```
